src/components/accordion/__tests__/CdrAccordion.spec.js

```diff
@@ -1,212 +1,158 @@
+    describe('when the accordion is closed', () => {
+      it('sets maxHeight to be 0', () => {
+        expect(wrapper.vm.maxHeight).toBe(0);
       });
 
-      expect(wrapper.vm.maxHeight).toBe(0);
-    });
-
-    it('maxHeight is adjusted in created when starting open', () => {
-      const wrapper = shallowMount(CdrAccordion, {
-        propsData: {
-          id: 'test',
-          level: '2',
-          opened: true,
-        },
-        slots: {
-          default: 'This is some slot text.',
-          label: 'label',
-        },
+      it('sets the "isOpenClass" computed prop to "cdr-accordion--closed" ', async () => {
+        expect(wrapper.vm.isOpenClass).toEqual('cdr-accordion--closed');
       });
-
-      expect(wrapper.vm.maxHeight).toBe('none');
-    });
-  });
-
-  describe('toggling accordion', () => {
-    it('updates emits an event onClick', async () => {
-      const wrapper = shallowMount(CdrAccordion, {
-        propsData: {
-          id: 'test',
-          level: '2',
-        },
-        slots: {
-          default: 'This is some slot text.',
-          label: 'label',
-        },
+      describe('when an accordion button receives focus', () => {
+        beforeEach(() => {
+          button.trigger('focus');
+        })
+        it('should be focused', ()=>{
+          expect(wrapper.vm.focused).toBeTruthy();
+        })
+        it('should have the proper focus class', ()=>{
+          expect(wrapper.classes()).toContain('cdr-accordion--focused');
+        })
+        describe('when an accordion button receives focus', () => {
+          beforeEach(() => {
+            button.trigger('blur');
+          });
+          it('should NOT be focused', ()=>{
+            expect(wrapper.vm.focused).toBeFalsy();
+          })
+          it('should NOT have the proper focus class', ()=>{
+            expect(wrapper.classes()).not.toContain('cdr-accordion--focused');
+          })
+        });
+      })
+      describe('a11y requirements', () => {
+        it('contains the "js-cdr-accordion-button" class', () => {
+          expect(button.classes()).toContain('js-cdr-accordion-button');
+        });
+
+        it('has the expected "aria-expanded" value', () => {
+          expect(button.attributes('aria-expanded')).toBe('false');
+        });
+
+        it('has the expected "aria-hidden" value for the content area', () => {
+          expect(contentArea.attributes('aria-hidden')).toBe('true');
+        });
+
+        it('has the expected "aria-controls: value', () => {
+          expect(button.attributes('aria-controls')).toBe(`${wrapper.vm.id}-collapsible`);
+        });
+      })
+    })
+
+    describe('when the accordion is open', () => {
+      beforeEach(async() => {
+        button.trigger('click');
+        wrapper.setProps({ opened: true }); // fake the opening logic
+        //TODO: e2e test where clicks work as expected. These could be visual regression tests where we check the visual appearance of the open state
```

Comment:
This is another case for an e2e test that we can add. We aren't testing the implementation of the click, just what happens when the `opened` prop changes. We're just assuming that a click will do that in this test.